.github/workflows/rust.yml

```
@@ -33,7 +33,7 @@ jobs:
       # before building the project.
       # e.g. icicle-cuda-runtime/src/bindings.rs is generated and icicle-cuda-runtime/src/lib.rs includes that module
       # causing rustfmt to fail.
-      run: if [[ $(find . -name target -prune -o -iname *.rs -print | xargs cargo fmt --check --) ]]; then echo "Please run cargo fmt"; exit 1; fi
+      run: if [[ $(cargo fmt --check --) ]]; then echo "Please run cargo fmt"; exit 1; fi
```

If this works then we should remove the comment above it regarding the need to check per file due to generated files not being present

```suggestion
      run: if [[ $(cargo fmt --check) ]]; then echo "Please run cargo fmt"; exit 1; fi
```

---

A bit out of scope here but still: do we want to extend fmt check to examples? I guess we'd need a matrix of working directories for that

---

sure, why not. Let's do it in some future PR